crates/accelerate/src/remove_identity_equiv.rs

```diff
@@ -83,42 +87,44 @@ fn remove_identity_equiv(
         let view = inst.op.view();
         match view {
             OperationRef::StandardGate(gate) => {
-                let (dim, trace) = match gate {
-                    StandardGate::RXGate | StandardGate::RYGate | StandardGate::RZGate => {
-                        if let Param::Float(theta) = inst.params_view()[0] {
-                            let trace = Complex64::new((theta / 2.).cos() * 2., 0.);
-                            (2., trace)
-                        } else {
-                            continue;
-                        }
-                    }
-                    StandardGate::RXXGate
+                let (tr_over_dim, dim) = match gate {
+                    StandardGate::RXGate
+                    | StandardGate::RYGate
+                    | StandardGate::RZGate
+                    | StandardGate::PhaseGate
+                    | StandardGate::RXXGate
                     | StandardGate::RYYGate
+                    | StandardGate::RZXGate
                     | StandardGate::RZZGate
-                    | StandardGate::RZXGate => {
-                        if let Param::Float(theta) = inst.params_view()[0] {
-                            let trace = Complex64::new((theta / 2.).cos() * 4., 0.);
-                            (4., trace)
+                    | StandardGate::CRXGate
+                    | StandardGate::CRYGate
+                    | StandardGate::CRZGate
+                    | StandardGate::CPhaseGate => {
+                        let name = gate.name();
+                        if let Param::Float(angle) = inst.params_view()[0] {
+                            let (tr_over_dim, dim) =
+                                rotation_trace_and_dim(name, angle).expect("All rotation covered");
```

Comment:
Since this becomes a panic message maybe something a bit more descriptive/clear. We should never encounter it because the match set is complete but since you went to the trouble of adding a message instead of just `unwrap()` we should provide a bit more detail.